x/bank/migrations/v043/store.go

```diff
@@ -75,9 +75,57 @@ func migrateBalanceKeys(store sdk.KVStore) {
 // - Change addresses to be length-prefixed.
 // - Change balances prefix to 1 byte
 // - Change supply to be indexed by denom
+// - Prune balances & supply with zero coins (ref: https://github.com/cosmos/cosmos-sdk/pull/9229)
 func MigrateStore(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec) error {
 	store := ctx.KVStore(storeKey)
 
+	if err := pruneZeroBalances(store, cdc); err != nil {
+		return err
+	}
+
+	if err := migrateSupply(store, cdc); err != nil {
+		return err
+	}
+
 	migrateBalanceKeys(store)
```

Comment:
We need to run `migrateBalanceKeys` before `pruneZeroBalances`, because pruneZeroBalances uses `balancesStore := prefix.NewStore(store, BalancesPrefix)` (new BalancesPrefix). Or else the iterator will always be empty.

Reply:
> We need to run migrateBalanceKeys before pruneZeroBalances

yes, thanks for finding this.